rustls-mbedcrypto-provider/tests/common/mod.rs

```diff
@@ -418,13 +427,47 @@ pub fn make_client_config(kt: KeyType) -> ClientConfig {
     )
 }
 
+static ALL_KEY_EXCHANGE_ALGORITHMS: &[KeyExchangeAlgorithm] = &[KeyExchangeAlgorithm::ECDHE, KeyExchangeAlgorithm::DHE];
+
+fn supported_kx_algos(provider: &CryptoProvider) -> Vec<KeyExchangeAlgorithm> {
+    let mut res = Vec::with_capacity(ALL_KEY_EXCHANGE_ALGORITHMS.len());
+    for kx in provider
+        .kx_groups
+        .iter()
+        .map(|kx| kx.name().key_exchange_algorithm())
+    {
+        if !res.contains(&kx) {
+            res.push(kx);
+        }
+        if res.len() == ALL_KEY_EXCHANGE_ALGORITHMS.len() {
+            break;
+        }
+    }
+    res
+}
+
+/// Return a `CryptoProvider` that have not matching `SupportedKxGroup`s in `kx_groups` removed.
+pub fn crypto_provider_without_matching_kx(mut provider: CryptoProvider) -> CryptoProvider {
```

Comment:
Nit: can this be renamed to something like `crypto_provider_with_cs_with_no_matching_kx_removed`?

Reply:
sure!